core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -203,7 +203,7 @@ public void save() throws IOException {
         SaveableListener.fireOnChange(this, config);
     }
 
-    public Object readResolve() {
+    private Object readResolve() {
```

Comment:
Sounds weird, but formally it's a compatibility breaking change. It could be a real case when you override the method in the inherited class, but this class is actually final. So I'm not sure it is important.

I would just add Restricted

Reply:
I do not think we care about formal compatibility when there is no plausible use case.

Reply:
Yes, but "no plausible use case" is unfortunately not the same as "Nobody is doing it and it won't break things."   

I tend to prefer relying on the Restricted modifier -- it'll break builds of plugins reliant on APIs that they shouldn't have access to, but doesn't cause direct binary-compat breakages.

Reply:
Or alternately, `@Restricted` for one LTS to show if this raises plugin issues, then private. 
This is a gentler way of doing binary-compat breaking changes IMO.  

In this case it's probably safe to break since, uh, readResolve... but in general we've been bitten many times when we said "oh there's no user" (and in one case recently even when an API was Restricted due to a quirk).